kategory-core/src/main/kotlin/kategory/data/NonEmptyList.kt

```diff
@@ -25,9 +25,9 @@ class NonEmptyList<out A> private constructor(
 
     val size: Int = all.size
 
-    fun contains(element: @UnsafeVariance A): Boolean = (head == element).or(tail.contains(element))
+    fun contains(element: @UnsafeVariance A): Boolean = (head == element) || element in tail
```

Comment:
@kategory/maintainers This (`||` instead of `or`) would avoid iterating through the tail if the element is in the head already.